src/test/regress/expected/multi_follower_dml.out

```diff
@@ -281,11 +282,14 @@ DELETE FROM the_table;
 ERROR:  cannot assign TransactionIds during recovery
 -- DDL is not possible
 TRUNCATE the_table;
-ERROR:  cannot execute TRUNCATE TABLE in a read-only transaction
+ERROR:  cannot acquire lock mode AccessExclusiveLock on database objects while recovery is in progress
```

Comment:
This is unfortunately a worse error message. But, I cannot think of an improvement at this point. It is at least not too bad an error message.

With MX, instead of sending `TRUNCATE ..`, we first send `lock_relation_if_exists()`.  Hence the message changes.

One thing we can do is `lock_relation_if_exists`  becomes no-op when read-only mode for `RowExclusive+` . But, I mean, that sounds weird..

So, we can go ahead and merge this